src/main/java/hillel/spring/doctor/DoctorRestAPIController.java

```diff
@@ -0,0 +1,99 @@
+
+
+import hillel.spring.doctor.dto.DoctorDtoConverterImpl;
+import hillel.spring.doctor.dto.DoctorInputDto;
+import hillel.spring.doctor.dto.DoctorOutputDto;
+import lombok.val;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static org.springframework.http.HttpStatus.NO_CONTENT;
+
+
+@RestController
+public class DoctorRestAPIController {
+    private final DoctorRestAPIService doctorRestAPIService;
+    private final DoctorDtoConverterImpl doctorDtoConverter;
+    private final SpecializationsConfig specializationsConfig;
+    private final UriComponentsBuilder uriComponentsBuilder;
+
+    public DoctorRestAPIController(DoctorRestAPIService doctorRestAPIService,
+                                   DoctorDtoConverterImpl doctorDtoConverter,
+                                   SpecializationsConfig specializationsConfig,
+                                   @Value("${doctor.hostname:localhost}") String hostname) {
+        this.doctorRestAPIService = doctorRestAPIService;
+        this.doctorDtoConverter = doctorDtoConverter;
+        this.specializationsConfig = specializationsConfig;
+        this.uriComponentsBuilder = UriComponentsBuilder.newInstance()
+                .scheme("http")
+                .host(hostname)
+                .port(8081)
+                .path("/doctors/{id}");
+    }
+
+    @GetMapping("/doctors")
+    public List<DoctorOutputDto> findDoctors(@RequestParam Optional<String> specialization,
+                                             @RequestParam Optional<String> name,
+                                             @RequestParam Optional<List<String>> specializations){
+
+        return doctorRestAPIService.findDoctors(specialization, name, specializations).stream()
+                                                          .map(doctorDtoConverter::toDto)
+                                                          .collect(Collectors.toList());
+    }
+
+    @GetMapping("/doctors/{id}")
+    public DoctorOutputDto findDoctorByID(@PathVariable Integer id){
+        val mayBeDoctor = doctorRestAPIService.findDoctorByID(id);
+
+        return doctorDtoConverter.toDto(mayBeDoctor.orElseThrow(DoctorNotFoundException::new));
+    }
+
+    @PostMapping("/doctors")
+    public ResponseEntity<?> createDoctor(@RequestBody DoctorInputDto docDto){
+        final val doctor = doctorDtoConverter.toModel(docDto);
+
+        if(checkSpecialization(doctor)) {
+            try {
+                final val newDoc = doctorRestAPIService.createDoctor(doctor);
+                return ResponseEntity.created(uriComponentsBuilder.build(newDoc.getId())).build();
+            } catch (Exception e) {
+                throw new RuntimeException("Oops, error");
```

Comment:
It is a really bad idea to swallow original exception: it contains a message and a stack trace. They can help find out what gone wrong.
Another bad idea to use this "Oops, error" it says nothing about the root cause, and it is better to avoid such messages.
So you should either wrap original exception or use lombok's `@SneakyThrows`  annotation